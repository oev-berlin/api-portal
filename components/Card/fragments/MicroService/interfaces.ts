interface MicroServiceProps {
    serviceName: string,
    isMicroservice: boolean,
}

export type { MicroServiceProps };
