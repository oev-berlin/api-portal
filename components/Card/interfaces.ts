import { ProjectData } from '../../utils/interfaces';

interface CardProps{
    projectData: ProjectData
}

export type { CardProps };
