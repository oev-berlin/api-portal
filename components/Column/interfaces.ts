import {projectData} from "../../utils/interfaces";

interface columnData {
    name: string,
    projects: projectData[],
}

export {columnData};