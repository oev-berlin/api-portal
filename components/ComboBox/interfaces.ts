export interface ComboBoxProps{
    searchOptions: string[],
}
