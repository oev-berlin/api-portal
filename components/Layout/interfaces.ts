import React from 'react';

export interface LayoutProps {
    searchOptions: string[],
    children: React.ReactNode,
}
