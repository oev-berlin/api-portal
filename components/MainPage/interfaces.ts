type docType = "backend" | "microservice";

interface projectData {
    id: string,
    name: string,
    description: string,
    swaggerInformation: object,
    docsType: docType,
    microservices: string[],
    externalServices: string[]
}

interface columnData {
    name: string,
    projects: JSON[]
}

export {projectData, columnData};

