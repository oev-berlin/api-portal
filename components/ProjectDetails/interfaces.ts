interface ProjectDetailsProps {
    name: string|undefined,
    description?: string|undefined,
}

export type { ProjectDetailsProps };
