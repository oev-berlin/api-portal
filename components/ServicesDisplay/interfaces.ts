interface ServiceDisplayProps {
    title: string,
    services: string[] | undefined,
}
export type { ServiceDisplayProps };
