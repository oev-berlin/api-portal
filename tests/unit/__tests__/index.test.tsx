import React from 'react';
import { cleanup, render, screen } from '@testing-library/react';
import Home from '../../../pages';
import { ProjectInfo } from '../../../utils/interfaces';
import { createProjectData } from '../../../utils/testUtilities';

const testProject1: ProjectInfo = createProjectData({
  id: '1',
  name: 'Test Project1',
  docsType: 'microservice',
});
const testProject2: ProjectInfo = createProjectData({
  id: '2',
  name: 'Test Project2',
  docsType: 'backend',
});

const setupComponent = ({ projects }: {projects: ProjectInfo[] }) => render(<Home projectsData={projects} />);

describe('Home', () => {
  afterEach(() => {
    cleanup();
  });

  describe('UI Tests', () => {
    it('Home component (with both backend and microservice projects) should render without crashing', async () => {
      const screen = setupComponent({
        projects: [testProject1, testProject2],
      });
      expect(screen.container).toBeDefined();
    });
    it('Home component (without microservice projects) should render without crashing', async () => {
      const screen = setupComponent({
        projects: [testProject2],
      });
      expect(screen.container).toBeDefined();
    });
    it('Home component (without backend projects) should render without crashing', async () => {
      const screen = setupComponent({
        projects: [testProject1],
      });
      expect(screen.container).toBeDefined();
    });
    it('Home component (without any project) should render without crashing', async () => {
      const screen = setupComponent({
        projects: [],
      });
      expect(screen.container).toBeDefined();
    });
  });
  it('Home component should render correctly', async () => {
    const screen = setupComponent({ projects: [testProject1, testProject2] });
    await screen
      .findByText(/microservices/i)
      .then((element) => expect(element).toBeInTheDocument());
    await screen
      .findByText(/backend/i)
      .then((element) => expect(element).toBeInTheDocument());
  });

  describe('Snapshots Tests', () => {
    it('should match a basic snapshot (with backend and microservice projects)', () => {
      const { container } = setupComponent({ projects: [testProject1, testProject2] });
      expect(container).toMatchSnapshot();
    });
    it('should match a basic snapshot (without microservice projects)', () => {
      const { container } = setupComponent({ projects: [testProject2] });
      expect(container).toMatchSnapshot();
    });
    it('should match a basic snapshot (without backend projects)', () => {
      const { container } = setupComponent({ projects: [testProject1] });
      expect(container).toMatchSnapshot();
    });
    it('should match a basic snapshot (without any project)', () => {
      const { container } = setupComponent({ projects: [] });
      expect(container).toMatchSnapshot();
    });
  });
});
