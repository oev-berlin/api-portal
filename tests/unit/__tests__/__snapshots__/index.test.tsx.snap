// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Snapshots Tests should match a basic snapshot (with backend and microservice projects) 1`] = `
<div>
  <div
    class="styles__MainPageOuterContainer-sc-138x4mg-0 bUphQo MuiBox-root css-0"
  >
    <h1
      class="styles__MainPageTitle-sc-138x4mg-2 bcHjKy"
    >
      Swagger API
    </h1>
    <div
      class="MuiGrid-root MuiGrid-container styles__MainPageInnerContainer-sc-138x4mg-1 bnqOnA css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 styles__ColumnContainer-sc-1b6u93i-0 eZXreS css-18kb45e-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
        >
          <h2
            class="styles__ColumnTitle-sc-1b6u93i-1 exaEdV"
          >
            Backend
          </h2>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 styles__ColumnContainer-sc-1b6u93i-0 eZXreS css-18kb45e-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
        >
          <h2
            class="styles__ColumnTitle-sc-1b6u93i-1 exaEdV"
          >
            Microservices
          </h2>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Home Snapshots Tests should match a basic snapshot (without any project) 1`] = `
<div>
  <div
    class="styles__MainPageOuterContainer-sc-138x4mg-0 bUphQo MuiBox-root css-0"
  >
    <h1
      class="styles__MainPageTitle-sc-138x4mg-2 bcHjKy"
    >
      Swagger API
    </h1>
    <div
      class="MuiGrid-root MuiGrid-container styles__MainPageInnerContainer-sc-138x4mg-1 bnqOnA css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 styles__ColumnContainer-sc-1b6u93i-0 eZXreS css-18kb45e-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
        >
          <h2
            class="styles__ColumnTitle-sc-1b6u93i-1 exaEdV"
          >
            Backend
          </h2>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 styles__ColumnContainer-sc-1b6u93i-0 eZXreS css-18kb45e-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
        >
          <h2
            class="styles__ColumnTitle-sc-1b6u93i-1 exaEdV"
          >
            Microservices
          </h2>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Home Snapshots Tests should match a basic snapshot (without backend projects) 1`] = `
<div>
  <div
    class="styles__MainPageOuterContainer-sc-138x4mg-0 bUphQo MuiBox-root css-0"
  >
    <h1
      class="styles__MainPageTitle-sc-138x4mg-2 bcHjKy"
    >
      Swagger API
    </h1>
    <div
      class="MuiGrid-root MuiGrid-container styles__MainPageInnerContainer-sc-138x4mg-1 bnqOnA css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 styles__ColumnContainer-sc-1b6u93i-0 eZXreS css-18kb45e-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
        >
          <h2
            class="styles__ColumnTitle-sc-1b6u93i-1 exaEdV"
          >
            Backend
          </h2>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 styles__ColumnContainer-sc-1b6u93i-0 eZXreS css-18kb45e-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
        >
          <h2
            class="styles__ColumnTitle-sc-1b6u93i-1 exaEdV"
          >
            Microservices
          </h2>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Home Snapshots Tests should match a basic snapshot (without microservice projects) 1`] = `
<div>
  <div
    class="styles__MainPageOuterContainer-sc-138x4mg-0 bUphQo MuiBox-root css-0"
  >
    <h1
      class="styles__MainPageTitle-sc-138x4mg-2 bcHjKy"
    >
      Swagger API
    </h1>
    <div
      class="MuiGrid-root MuiGrid-container styles__MainPageInnerContainer-sc-138x4mg-1 bnqOnA css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 styles__ColumnContainer-sc-1b6u93i-0 eZXreS css-18kb45e-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
        >
          <h2
            class="styles__ColumnTitle-sc-1b6u93i-1 exaEdV"
          >
            Backend
          </h2>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 styles__ColumnContainer-sc-1b6u93i-0 eZXreS css-18kb45e-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
        >
          <h2
            class="styles__ColumnTitle-sc-1b6u93i-1 exaEdV"
          >
            Microservices
          </h2>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
        />
      </div>
    </div>
  </div>
</div>
`;
